veps/sig-storage/declarative-volume-hotplug.md

```diff
@@ -0,0 +1,352 @@
+
+## Definition of Users
+
+- End Users: these are people/programs that have permission to update Virtual Machine specifications
+
+## User Stories
+
+- As a KubeVirt user, I want to be able to use `kubectl edit` to add/remove/change the volumes of a Virtual Machine and have those changes take effect immedietly without restarting the VM
+- As a KubeVirt user, I want to be able to add/remove/change the volumes of a Virtual Machine by updating the VM definition, pushing it to a git repository, and when a system like [Open Cluster Management](https://open-cluster-management.io/) or [ArgoCD](https://argoproj.github.io/cd/) applies those changes, they will take effect immedietly without restarting the VM
+- As a KubeVirt user, I want to be able define a CD-ROM disk on a Virtual Machine and simulate injecting/ejecting a CD-ROM disk by adding/removing volumes from the VM specification. Those changes should take effect immedietly without starting the Virtual Machine
+
+## Repos
+
+kubevirt/kubevirt
+
+## Design
+
+### Enabling Declarative Volume Hotplug
+
+A couple different options were considered.
+
+#### Option 1 - LiveUpdate VM Rollout Strategy + Hotplug Volume Update Strategy
+
+This option leverages [VM Rollout Strategy](https://github.com/kubevirt/community/blob/main/design-proposals/vm-rollout-strategy/vm-rollout-strategy.md) and [Volume Update Strategy](https://github.com/kubevirt/community/blob/main/design-proposals/volume-update-strategy.md)
+
+- A new Volume Update Strategy called `HotPlug` will be created
+- Declarative Volume update will be enabled when the VM Rolout Strategy is `LiveUpdate` and Volume Update Strategy is `Hotplug`
+- If either of those values are not set, the volume update will not be applied until the Virtual Machine is restarted
+
+Pros:
+- Backward compatible
+- Users can opt in
+
+Cons:
+- Cannot achieve soft goal of having the subresource API leverage the declarative API
+
+#### Option 2 - New default (most of the time)
+
+In this case, the feature will be enabled by default except when VM Rollout strategy is `LiveUpdate` AND Volume Update Strategy is `Migration`
+
+There are currently two Volume Update Strategies, `Replacement` and `Migration`. `Replacement` "stages" declarative volume changes to be applied when the Virtual Machine restarts. `Migrtion` kicks off a live migration which will copy a volume's data to a new PVC.
+
+With this option, the `Replacement` strategy will live update changes to any volume with `hotpluggable: true`
+
+Pros:
+- Users can opt into this feature without regard to VM Rollout Strategy which is a cluster-scope setting
+- Soft goal of combining the declarative and subresource APIs can be achieved
+
+Cons:
+- Not backward compatible (can have a feature gate initially)
+- Can't hotplug volumes during storage migration unless we don't combine subresource and declarative APIs
```

Comment:
Volume migration works on set of volumes which cannot be change dynamically once it is triggered. This is why we try to restrict the volume migration to a single change. If you start to change the set of volumes to consider, then this becomes extremely difficult to interpret..
Hence, if a volume is added after the change which triggered the migration, then it means that you change the volume strategy to replacement +  the change for the hotplugged volumes. So, I won't see this as a cons

Reply:
Got it. So once a storage migration is kicked off, it will complete regardless of the update strategy. I will remove this con.

@alicefr sounds like you may be leaning towards option 2?